src/jit/MemoryInl.h

```diff
@@ -1011,3 +1011,356 @@ static void emitStore(sljit_compiler* compiler, Instruction* instr)
+    case ByteCode::I64AtomicRmw16AndUOpcode:
+    case ByteCode::I64AtomicRmw32AndUOpcode: {
+        operation = SLJIT_AND;
+        break;
+    }
+    case ByteCode::I32AtomicRmwOrOpcode:
+    case ByteCode::I32AtomicRmw8OrUOpcode:
+    case ByteCode::I32AtomicRmw16OrUOpcode:
+    case ByteCode::I64AtomicRmwOrOpcode:
+    case ByteCode::I64AtomicRmw8OrUOpcode:
+    case ByteCode::I64AtomicRmw16OrUOpcode:
+    case ByteCode::I64AtomicRmw32OrUOpcode: {
+        operation = SLJIT_OR;
+        break;
+    }
+    case ByteCode::I32AtomicRmwXorOpcode:
+    case ByteCode::I32AtomicRmw8XorUOpcode:
+    case ByteCode::I32AtomicRmw16XorUOpcode:
+    case ByteCode::I64AtomicRmwXorOpcode:
+    case ByteCode::I64AtomicRmw8XorUOpcode:
+    case ByteCode::I64AtomicRmw16XorUOpcode:
+    case ByteCode::I64AtomicRmw32XorUOpcode: {
+        operation = SLJIT_XOR;
+        break;
+    }
+    case ByteCode::I32AtomicRmwXchgOpcode:
+    case ByteCode::I32AtomicRmw8XchgUOpcode:
+    case ByteCode::I32AtomicRmw16XchgUOpcode:
+    case ByteCode::I64AtomicRmwXchgOpcode:
+    case ByteCode::I64AtomicRmw8XchgUOpcode:
+    case ByteCode::I64AtomicRmw16XchgUOpcode:
+    case ByteCode::I64AtomicRmw32XchgUOpcode: {
+        operation = OP_XCHG;
+        break;
+    }
+    default:
+        RELEASE_ASSERT_NOT_REACHED();
+        break;
+    }
+
+    AtomicRmw* rmwOperation = reinterpret_cast<AtomicRmw*>(instr->byteCode());
+    offset = rmwOperation->offset();
+
+    Operand* operands = instr->operands();
+    MemAddress addr(options, instr->requiredReg(0), instr->requiredReg(1), instr->requiredReg(2));
+
+#if (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+    JITArgPair valueArgPair;
+    if ((operationSize & SLJIT_32) || operationSize == SLJIT_MOV32) {
+        addr.loadArg.set(operands + 1);
+
+        if (SLJIT_IS_MEM(addr.loadArg.arg)) {
+            addr.options |= MemAddress::LoadInteger;
+        }
+    } else {
+        valueArgPair.set(operands + 1);
+
+        addr.loadArg.arg = valueArgPair.arg1;
+        addr.loadArg.argw = valueArgPair.arg1w;
+
+        if (SLJIT_IS_MEM(valueArgPair.arg1)) {
+            addr.options |= MemAddress::LoadInteger;
+        }
+    }
+#else /* !SLJIT_32BIT_ARCHITECTURE */
+    addr.loadArg.set(operands + 1);
+
+    if (SLJIT_IS_MEM(addr.loadArg.arg)) {
+        addr.options |= MemAddress::LoadInteger;
+
+        if (operationSize == SLJIT_MOV32 || (operationSize & SLJIT_32)) {
+            addr.options |= MemAddress::Load32;
+        }
+    }
+#endif /* SLJIT_32BIT_ARCHITECTURE */
+
+    addr.check(compiler, operands, offset, size);
+
+    JITArg dst;
+    sljit_s32 srcReg;
+
+#if (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+    if ((operationSize & SLJIT_32) || operationSize == SLJIT_MOV32) {
+#endif /* SLJIT_32BIT_ARCHITECTURE */
+        JITArg src(operands + 1);
+        dst = JITArg(operands + 2);
+        srcReg = GET_SOURCE_REG(src.arg, instr->requiredReg(1));
```

Comment:
I don't understand something. SrcReg is set to `instr->requiredReg(1)`. 

Here sourceReg is the last argument, which is set to ` instr->requiredReg(2)`.
https://github.com/Samsung/walrus/blob/main/src/jit/MemoryInl.h#L37